src/Features/VisualBasic/Portable/Completion/CompletionProviders/SymbolCompletionProvider.vb

```diff
@@ -88,10 +88,6 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Completion.Providers
             For importDirective = 0 To 1
                 For preselect = 0 To 1
                     For tuple = 0 To 1
-                        If importDirective = 1 AndAlso tuple = 1 Then
-                            Continue For
-                        End If
-
                         Dim context = ValueTuple.Create(importDirective = 1, preselect = 1, tuple = 1)
```

Comment:
What even is this code?

Reply:
There are 8 possible completion rules, depending on the context. See MakeRules method.
This loop makes them and caches them.